src/server.cpp

```diff
@@ -3797,6 +3798,9 @@ PlayerSAO* Server::emergePlayer(const char *name, session_t peer_id, u16 proto_v
 	playersao->finalize(player, getPlayerEffectivePrivs(player->getName()));
 	player->protocol_version = proto_version;
 
+	// Set player lang_code
+	player->setLangCode(getClient(peer_id, CS_Invalid)->getLangCode());
```

Comment:
Passing the language from `RemoteClient` to the `RemotePlayer` is kinda more complicated than necessary.

The server could expose a method to read the language of a player and the API implementation can then retrieve it similarly to how `l_get_player_information` in l_server.cpp works.

Reply:
In that case it could even be simpler to add lang_code to information retreived by `l_get_player_information` instead of the new `get_player_language` method.
This would make things a little bit more complicated for modders but anyway, such use of server side translations is very specific.